Find_first_missing_positive_integer.rb

```diff
@@ -0,0 +1,29 @@
+=begin 
+This problem was asked by Stripe.
+
+Given an array of integers, find the first missing positive integer in linear time and constant space. 
+In other words, find the lowest positive integer that does not exist in the array. 
+The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+=end 
+
+def missingpostiveint(array)
+    #array = array.sort 

```

Comment:
this line needs to be commented in? we need the array to be sorted for this to work